.infrastructure/helm-chart/todoapp/templates/hpa.yml

```diff
@@ -0,0 +1,25 @@
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{.Chart.Name}}
+  minReplicas: {{.Values.todoapp.hpa.minReplicas}}
+  maxReplicas: {{.Values.todoapp.hpa.maxReplicas}}
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: {{.Values.todoapp.hpa.resourceAvgUtilization.cpu}}
+  - type: Resource
+    resource:
+      name: memory
+      target:
+        type: Utilization
+        averageUtilization: {{.Values.todoapp.hpa.resourceAvgUtilization.memory}}
```

Comment:
Ensure that `{{.Values.todoapp.hpa.resourceAvgUtilization.memory}}` is defined in your `values.yaml` file. If it's not defined, it could lead to autoscaling issues.